src/evagram/database/input_tool.py

```diff
@@ -0,0 +1,174 @@
+import pickle
+import os
+import psycopg2
+
+
+class Session:
+    def __init__(self, conn, wf_dictionary):
+        self.conn = conn
+        self.wf_dictionary = wf_dictionary
+        self.owner_id = None
+        self.experiment_id = None
+        self.open_session()
+
+    def open_session(self):
+        # Verifying expected parameters
+        required = {'owner', 'experiment', 'eva_directory'}
+        difference = required.difference(self.wf_dictionary)
+        if len(difference) > 0:
+            print(f"Missing required parameters for workflow dictionary: {difference}")
+            return 1
+        with self.conn:
+            with self.conn.cursor() as cur:
+                cur.execute("SELECT pg_backend_pid();")
+                conn_pid = cur.fetchone()[0]
+                cur.execute("SELECT usename FROM pg_stat_activity WHERE pid=%s", (conn_pid,))
+                conn_username = cur.fetchone()[0]
+                if conn_username != self.wf_dictionary["owner"]:
+                    print("Terminating session, invalid workflow parameters.")
+                    return 1
+                self.owner_id = add_current_user(cur, self.wf_dictionary["owner"])
+                self.experiment_id = add_current_experiment(
+                    cur, self.wf_dictionary["experiment"], self.owner_id)
+                print("Session created successfully! Running task...")
+                self.run_task()
+                print("Task completed!")
+        return 0
+
+    def run_task(self):
+        eva_directory_path = self.wf_dictionary["eva_directory"]
+        with self.conn.cursor() as cur:
+            for observation_dir in os.listdir(eva_directory_path):
+                observation_path = os.path.join(eva_directory_path, observation_dir)
+                if os.path.isdir(observation_path):
+                    for plot in os.listdir(observation_path):
+                        if plot.endswith(".pkl"):
+                            add_plot(
+                                cur, eva_directory_path, observation_dir, plot, self.experiment_id)
+        return 0
+
+
+def insert_table_record(cur, data, table):
```

Comment:
OR, another option is for the `insert` and `add` functions to remain external but to accept a Session object instead of the cursor.  My  main point is, the cursor should be protected and  unexposed.